packages/flutter/lib/src/material/text_selection_theme.dart

```diff
@@ -0,0 +1,167 @@
+  final Color selectionColor;
+
+  /// The color used to paint the selection handles on the text field.
+  ///
+  /// Selection handles are used to indicate the bounds of the selected text,
+  /// or as a handle to drag the cursor to a new location in the text.
+  final Color selectionHandleColor;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// specified values.
+  TextSelectionThemeData copyWith({
+    Color cursorColor,
+    Color selectionColor,
+    Color selectionHandleColor,
+  }) {
+    return TextSelectionThemeData(
+      cursorColor: cursorColor ?? this.cursorColor,
+      selectionColor: selectionColor ?? this.selectionColor,
+      selectionHandleColor: selectionHandleColor ?? this.selectionHandleColor,
+    );
+  }
+
+  /// Linearly interpolate between two text field themes.
+  ///
+  /// If both arguments are null, then null is returned.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static TextSelectionThemeData lerp(TextSelectionThemeData a, TextSelectionThemeData b, double t) {
+    if (a == null && b == null)
+      return null;
+    assert(t != null);
+    return TextSelectionThemeData(
+      cursorColor: Color.lerp(a?.cursorColor, b?.cursorColor, t),
+      selectionColor: Color.lerp(a?.selectionColor, b?.selectionColor, t),
+      selectionHandleColor: Color.lerp(a?.selectionHandleColor, b?.selectionHandleColor, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      cursorColor,
+      selectionColor,
+      selectionHandleColor,
+    );
+  }
+
+  @override
+  bool operator==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is TextSelectionThemeData
+      && other.cursorColor == cursorColor
+      && other.selectionColor == selectionColor
+      && other.selectionHandleColor == selectionHandleColor;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ColorProperty('cursorColor', cursorColor, defaultValue: null));
+    properties.add(ColorProperty('selectionColor', selectionColor, defaultValue: null));
+    properties.add(ColorProperty('selectionHandleColor', selectionHandleColor, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the configuration for
+/// the rendering of text selection in this widget's subtree.
+///
+/// Values specified here are used for [TextField] and [SelectableText]
+/// properties that are not given an explicit non-null value.
+///
+/// {@tool snippet}
+///
+/// Here is an example of a text selection theme that applies a blue cursor
+/// color with light blue selection handles to the child text field.
+///
+/// ```dart
+/// TextSelectionTheme(
+///   data: TextSelectionThemeData(
+///     cursorColor: Colors.blue,
+///     selectionHandleColor: Colors.lightBlue,
+///   ),
+///   child: TextField(),
```

Comment:
This works because the TextField doesn't actually use TextSelectionTheme directly right? Otherwise we'd have to include a Builder.

Reply:
`TextField` looks up the current theme with `TextSelectionTheme.of(context)` in the build method, so it should work fine here.